src/index.ts

```diff
@@ -0,0 +1,75 @@
+import express, { Request, Response, NextFunction} from 'express';
+
+import Datasource from './Datasource';
+import UserController from './user/user.controller';
+import ProductController from './product/product.controller';
+import CartController from './cart/cart.controller';
+import OrderController from './order/order.controller';
+
+import { UserRepository } from './user/user.repository';
+import { ProductRepository } from './product/product.repository';
+import { CartRepository } from './cart/cart.repository';
+import { OrderRepostiory } from './order/order.repository';
+
+import { UserService } from './user/user.service';
+
+const app = express();
+
+// Data init
+const database = {
+  users: Datasource.users,
+  products: Datasource.products,
+  orders: [],
+  carts: []
+}
+
+// Repositories init
+const userRepository = new UserRepository(database.users);
+const productRepository = new ProductRepository(database.products);
+const cartRepository = new CartRepository(database.carts);
+const orderRepository = new OrderRepostiory(database.orders);
+
+//Middlewares
+const userService = new UserService(userRepository);
+const authMiddleware = (req: Request, res: Response, next: NextFunction) => {
+  const userId = req.get('x-user-id');
+  if(userId == undefined) {
+    //	Forbidden(when authorization header is missing)
+    res.status(403).send({
```

Comment:
I'm not sure about the errors for the user header, in the Swagger it wasn't clear enough.
Let me know if the error handling was ok, or if I need to make some changes.

Reply:
You are right, the swagger and task description are different. The lets say both ways are correct.